python/katana/bug/environment.py

```diff
@@ -0,0 +1,227 @@
+import datetime
+import os
+import re
+import socket
+import subprocess
+import sys
+import tempfile
+import zipfile
```

Comment:
A small preference for using `tarfile` as our defacto bag of files format as it supports Unixy features more directly like UID and special files if we ever need them later.

Reply:
I used zip because there is better support for looking at files inside it from many OSs and software. Some, notably Windows Explorer, don't have tar support (I assume this has not changed recently). I would generally prefer tar as well, but I decided zip would be easier for more people to look at and this archive does not need to be unpacked and be runnable or anything.

Is that a good enough reason for you? I can go with tar if you don't think this is a good argument.

Reply:
Is this something we expect reporters/bug submitters to unpack?

Anyway, if you can imagine a common workflow that would require windows users to unpack a report, zip is reasonable. Myself, I don't know what that would be.

Reply:
The workflow I see is this: We have a lot of developers who use windows as their main OS and use a VM or remote host with Linux. They don't generally have their web browser open in that Linux environment (especially for people without a local Linux VM), so developers working on a bug will download it on to their local windows machine and want to look at it. It would be an extra step to copy the report to a linux machine and then extract on the remote host.

Reply:
FWIW, I think most windows developers would use WSL if that matters and Windows 10 comes with tar, the command line utility.

To build or run anything in this repo, you probably need a typical Linux environment, I don't know how much extra variety we should support, especially when we dictate what constitutes a "valid developer environment".

Anyway, I don't think consistency here is a big thing one way or another.

Reply:
Makes sense. This can be changes easily if needed. I'm not wedded to one decision and changing it later wouldn't really break anything.

Now I'm curious how peoples dev environments are setup. I wonder how many people have their Linux dev work and their web browser accessing the same file system and how many do not. Hmm.